### PR TITLE
skip doc only travis builds

### PR DESCRIPTION
**Pull request by vegichan**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,11 +22,20 @@ env:
...
 
 before_install:
+  - |
+    set -e
+    # fail loudly when force-pushed
+    MODIFIED_FILES=$(git diff --name-only $TRAVIS_COMMIT_RANGE)
+    # waiting for native solution https://github.com/travis-ci/travis-ci/issues/6301
+    if ! echo ${MODIFIED_FILES} | grep -qvE '(\.md$)'; then
+      echo "Only docs were updated, stopping build process."
+      exit
```
Comment (thebithead) [inline .travis.yml]: I'm wondering if you should call `travis_terminate 0` here instead of just calling exit.

File: .travis.yml [hunk 2]
```diff
@@ -22,11 +22,20 @@ env:
     - GO111MODULE=on
 
 before_install:
+  - |
+    set -e
+    # fail loudly when force-pushed
+    MODIFIED_FILES=$(git diff --name-only $TRAVIS_COMMIT_RANGE)
```
Comment (thebithead) [inline .travis.yml]: The Travis-CI doc says that $TRAVIS_COMMIT_RANGE will be empty for builds triggered by the initial commit of a new branch.

Comment (vegichan) [reply .travis.yml]: hmm, what is the alternative? `diff master`?

Comment (thebithead) [reply .travis.yml]: One alternative is to allow the build to continue if MODIFIED_FILES is empty.

Comment (vegichan) [reply .travis.yml]: what do you think of this

`if [ -z "${MODIFIED_FILES}" ]; then
  # $TRAVIS_COMMIT_RANGE will be empty for builds triggered by the initial commit of a new branch.
  echo "No changes found, can not determine what to skip"
elif ! echo ${MODIFIED_FILES} | grep -qvE '(\.md$)'; then
  echo "Only docs were updated, stopping build process."
  exit
fi`

Comment (thebithead) [reply .travis.yml]: I think that would work.

File: .travis.yml (hunk 1 above)
Comment (vegichan) [reply .travis.yml]: I will test this once it is in. If it does not work as expected, we can rethink. I can not test this in a branch

Comment (thebithead) [reply .travis.yml]: Why can you test this change in your Travis-CI branch build?

Comment (vegichan) [reply .travis.yml]: works as expected

https://travis-ci.org/github/nuodb/nuodb-operator/jobs/663202755?utm_medium=notification&utm_source=github_status

**Pull request merged**